schema.go

```diff
@@ -349,10 +360,40 @@ type structNode struct {
 	fields []structField
 }
 
-func structNodeOf(t reflect.Type) *structNode {
+// StructTag contains struct tag values that are use by parquet-go to configure
+// nodes.
+type StructTag struct {
+	// Value set on parquet struct tag
+	Parquet string
+	// Value set on parquet-key struct tag
+	MapKey string
+	// Value set on parquet-value struct tag
+	MapValue string
+}
+
+type TagSource interface {
+	Tags(f reflect.StructField) StructTag
+	isTagSource()
+}
+
+type defaultTagSource struct{}
+
+var _ TagSource = defaultTagSource{}
+
+func (defaultTagSource) Tags(f reflect.StructField) StructTag {
+	return StructTag{
+		Parquet:  f.Tag.Get("parquet"),
+		MapKey:   f.Tag.Get("parquet-key"),
+		MapValue: f.Tag.Get("parquet-value"),
+	}
+}
+
+func (defaultTagSource) isTagSource() {}
```

Comment:
The top-level package is already very populated, and since these types do not really have anything to do with Parquet, what do you think about moving them to a `schema` package?

This could look like:
```go
package schema

type Option ...

type StructTag ...

type TagSource ...
```

Reply:
sounds good, done 095ea7d